assets/vue/components/PageBlocked.vue

```diff
@@ -54,21 +62,41 @@ export default {
       incidents: null,
       groupFlavors: true,
       matchText: '',
+      groupNames: '',
       refreshUrl: '/app/api/blocked'
     };
   },
   computed: {
     matchedIncidents() {
+      let results = [];
       if (this.matchText) {
-        return this.incidents.filter(incident => {
+        results = this.incidents.filter(incident => {
           if (String(incident.incident.number).includes(this.matchText)) return true;
           for (const pack of incident.incident.packages) {
             if (pack.includes(this.matchText)) return true;
           }
           return false;
         });
+      } else {
+        results = this.incidents;
+      }
+      if (this.groupNames) {
+        return results.filter(incident => {
+          const groupNamesList = this.groupNames.toLowerCase().split(',');
```

Comment:
This looks like it should be outside the closure.

Reply:
Yes - it makes no sense to compute this list on every filter callback.